### PR TITLE
Fix/lp 1838782

### PR DESCRIPTION
**Pull request by ycliuhw**
## Description of change

fix `--force` broken issue on `destroy-model` cmd;

driveby fix: exit peacefully if `RemoteStateWatcher` got `not found` error(application has already been removed in this case);

## QA steps

1. deploy workload;

```bash
$ juju add-model t1 microk8s && juju deploy cs:~kelvin.liu/mariadb-k8s-3
```

2. destroy model with `--force`;

```bash
$ juju destroy-model t1 --destroy-storage -y --debug --force
```

## Documentation changes

None

## Bug reference

https://bugs.launchpad.net/juju/+bug/1838782

### CODE REVIEW

**Comment by ycliuhw** [comment]
[https://bugs.launchpad.net/juju/+bug/1838782](https://trello.com/c/UNDOdy9f/880-https-bugslaunchpadnet-juju-bug-1838782)

File: worker/caasoperator/remotestate/watcher.go [hunk 1]
```diff
@@ -81,6 +82,12 @@ func (w *RemoteStateWatcher) Snapshot() Snapshot {
 }
 
 func (w *RemoteStateWatcher) loop() (err error) {
+	defer func() {
+		if errors.IsNotFound(err) {
+			err = jworker.ErrTerminateAgent
```
Comment (wallyworld) [inline worker/caasoperator/remotestate/watcher.go]: Only IAAS models can have the agent terminated. With k8s models, all the uniters run as workers and we need to keep the agent running. For k8s uniter workers, we simple exit the worker.

Comment (ycliuhw) [reply worker/caasoperator/remotestate/watcher.go]: ic, probably we can return `jworker.ErrTerminateAgent` for application not found?

Comment (ycliuhw) [reply worker/caasoperator/remotestate/watcher.go]: yeah, changed to exit peacefully if it got a not found error.

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**